Add test 'Add-delete Ironic node to cluster'

gdyuldin opened the pull request:
This test case checks ironic work after adding new ironic-conductor node
to exist cluster.

gdyuldin committed:
Add test 'Add new Ironic node to cluster'

This testcase checks ironic work after adding new ironic-conductor node
to exist cluster.

gdyuldin committed:
Extend test 'Add new Ironic node to cluster'

This patch adds ironic+controller and ironic+controller+ceph roles
parameters of test scenario.

schipiga commented on mos_tests/__init__.py:
```diff
@@ -21,12 +21,32 @@
+    waiting_fmt = '%(asctime)s [%(levelname)s] %(message)s'
+
+    def __init__(self, fmt):
+        super(WaitingFormatter, self).__init__(fmt)
```
hmm, isn't it redundant?

schipiga commented on mos_tests/functions/common.py:
```diff
@@ -600,12 +601,19 @@ def delete_keys(nova_client, key_name):
-        logger.info('waiting for {} ... done'.format(waiting_for))
+        logger.info(msg + ' ... done')
         return result
     except TimeoutExpired as e:
```
that looks strange for me. Can't understand why need to catch exception, if then it is raised again.

gdyuldin replied:
It was made to prevent py.test shows internal calls of `waiting` module if `TimeoutExpired` raises.

kromanenko commented on mos_tests/ironic/actions.py:
```diff
@@ -0,0 +1,129 @@
+                                          timeout=60 * 10,
+                                          **kwargs)
+
+    def create_node(self, baremetal_node, server_ssh_credentials):
```
Possibly good to extend with creation of node with IPMI creds.

gdyuldin replied:
@kromanenko good idea. As i known, you use yaml configuration file for IPMI in fuel-qa. Can you provide an example of configuration (only structure)?

gdyuldin committed:
Add support of IPMI (or another) Ironic driver

gdyuldin committed:
Add tests 'Delete one of multiple ironic nodes'

This tests delete created on previous step ironic nodes and checks that
ironic works.
